src/frog.js

```diff
@@ -0,0 +1,360 @@
+const BtpPacket = require('btp-packet')
+const IlpPacket = require('ilp-packet')
+const crypto = require('crypto')
+
+function generateRequestId () {
+  const buf = crypto.randomBytes(4)
+  const hex = buf.toString('hex')
+  return parseInt(hex, 16)
+}
+
+function MakeProtocolData (obj) {
+  let protocolData = []
+  if (obj.from) {
+    protocolData.push({
+      protocolName: 'from',
+      contentType: BtpPacket.MIME_TEXT_PLAIN_UTF8,
+      data: Buffer.from(obj.from, 'ascii')
+    })
+  }
+  if (obj.to) {
+    protocolData.push({
+      protocolName: 'to',
+      contentType: BtpPacket.MIME_TEXT_PLAIN_UTF8,
+      data: Buffer.from(obj.to, 'ascii')
+    })
+  }
+  if (obj.ilp) {
+    protocolData.push({
+      protocolName: 'ilp',
+      contentType: BtpPacket.MIME_APPLICATION_OCTET_STREAM,
+      data: Buffer.from(obj.ilp, 'base64')
+    })
+  } else {
+    protocolData.push({
+      protocolName: 'ccp',
+      contentType: BtpPacket.MIME_APPLICATION_JSON,
+      data: Buffer.from(JSON.stringify(obj.custom), 'ascii')
+    })
+  }
+  return protocolData
+}
+
+function Frog (plugin, send) {
+  this.plugin = plugin
+  this.send = send
+  this.registerPluginEventHandlers()
+  this.requestsReceived = {}
+}
+
+Frog.prototype = {
```

Comment:
Make this an ES6 class?

Reply:
Why?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes
> JavaScript classes, introduced in ECMAScript 2015, are primarily syntactical sugar over JavaScript's existing prototype-based inheritance. The class syntax is not introducing a new object-oriented inheritance model to JavaScript. JavaScript classes provide a much simpler and clearer syntax to create objects and deal with inheritance.

I'm not using inheritance here, so I think the only difference would be I could leave out the `,`
If that's really the only thing, then it's probably not worth it?

Reply:
I agree sugar can sometimes be useful, but we should also not put sugar on everything.

To understand the sugared version of the code, you still need to understand the underlying version, so if the visual syntax differences are not significant, then it's usually not useful (see also https://github.com/interledgerjs/btp-toolbox/pull/1#discussion_r139639672)